lib/core/mqtt.cpp

```diff
@@ -99,9 +99,7 @@ mqtt::message_ptr Mqtt::messageToMessagePtr(const MqttMessage &message)
   header.AddMember("priority", message.header.priority, payload.GetAllocator());
   header.AddMember("timestamp", message.header.timestamp, payload.GetAllocator());
   payload.AddMember("header", header, payload.GetAllocator());
-  rapidjson::Value payload_json;
-  payload_json.CopyFrom(message.payload, payload.GetAllocator());
-  payload.AddMember("payload", payload_json, payload.GetAllocator());
+  payload.CopyFrom(*message.payload, payload.GetAllocator());
```

Comment:
I believe this will overwrite the existing JSON
either call this before adding the header as a member, or by adding another member containing message.payload